src/auth.ts

```diff
@@ -1,15 +1,84 @@
 import * as vscode from "vscode";
 
-export const getApiKeyCredentials = () => {
-  const configurationWorkspace = vscode.workspace.getConfiguration();
-  const env0ApiKey: string | undefined =
-    configurationWorkspace.get("env0.apiKeyId");
-  const env0SecretKey: string | undefined =
-    configurationWorkspace.get("env0.secretKey");
+const env0KeyIdKey = "env0.keyId";
+const env0SecretKey = "env0.secret";
 
-  if (!env0ApiKey || !env0SecretKey) {
-    throw new Error("Could not read env0 api key values");
+export class AuthService {
+  public onAuth?: () => any = undefined;
+  constructor(private readonly context: vscode.ExtensionContext) {}
+  public registerLoginCommand() {
+    const disposable = vscode.commands.registerCommand(
+      "env0.login",
+      async () => {
+        const keyId = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: "API Key ID",
+          prompt: "Enter your API Key ID.",
+          password: true,
+          validateInput: (token) => {
+            if (!token) {
+              return "API Key ID cannot be empty";
+            }
+            return null;
+          },
+        });
+
+        const secret = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: "API Key Secret",
+          prompt: "Enter your API Key Secret.",
+          password: true,
+          validateInput: (token) => {
+            if (!token) {
+              return "API Key Secret cannot be empty";
+            }
+            return null;
+          },
+        });
+        await this.storeAuthData(keyId!, secret!);
+        this.onAuth?.();
+      }
+    );
+    this.context.subscriptions.push(disposable);
+  }
+
+  public registerLogoutCommand() {
+    const disposable = vscode.commands.registerCommand(
```

Comment:
what does `disposable` mean in this context?

Reply:
the idea is to cleanup all the listeners when extension deactivated
https://stackoverflow.com/questions/55554018/purpose-for-subscribing-a-command-in-vscode-extension